docs/source/execute_api.rst

```diff
@@ -0,0 +1,143 @@
+Executing notebooks
+===================
```

Comment:
Well the example give here is explicitly reading from disk, executing and writing back. So it is re-writing a basic exporter. Using the exporter also remove from many underlying details, like reading the notebook correctly (with nbformat) and setting the kernel to the right kerne, and then writing it back. 

Plus we know from experience that the order of preprocessors is tricky. So we likely do not want people to use preprocessor in the wrong way/wrong order and complain.